drivers/network/meson/ethernet.c

```diff
@@ -210,18 +210,24 @@ static void tx_return(void)
 static void handle_irq()
 {
     uint32_t e = eth_dma->status;
-    if (e & DMA_INTR_RXF) {
-        rx_return();
-    }
-    if (e & DMA_INTR_TXF) {
-        tx_return();
-    }
-    if (e & DMA_INTR_ABNORMAL) {
-        if (e & DMA_INTR_FBE) {
-            sddf_dprintf("Ethernet device fatal bus error\n");
+    eth_dma->status &= e;
+
+    while (e & DMA_INTR_MASK) {
+        if (e & DMA_INTR_RXF) {
+            rx_return();
         }
+        if (e & DMA_INTR_TXF) {
+            tx_return();
+            tx_provide();
+        }
+        if (e & DMA_INTR_ABNORMAL) {
+            if (e & DMA_INTR_FBE) {
+                sddf_dprintf("Ethernet device fatal bus error\n");
```

Comment:
Can you print out the value of `e` when we get the error.

Reply:
Actually maybe it's not useful to print it out - I guess it would tell you what the DMA status was at the time of the bus error which may help?

Reply:
Sure, I will add it to the print statements. Not sure how useful it will be (have a feeling that when `DMA_INTR_FBE` is set it is basically saying that something bad happened and the system is now in an unrecoverable state), but I am not entirely sure.